src/jit/codegencommon.cpp

```diff
@@ -7545,12 +7608,17 @@ void CodeGen::genProfilingLeaveCallback(unsigned helper /*= CORINFO_HELP_PROF_FC
         noway_assert((RBM_PROFILER_LEAVE_TRASH & thisPtrMask) == 0);
     }
 
-    // At this point return value is computed and stored in RAX or XMM0.
-    // On Amd64, Leave callback preserves the return register.  We keep
-    // RAX alive by not reporting as trashed by helper call.  Also note
-    // that GC cannot kick-in while executing inside profiler callback,
-    // which is a requirement of profiler as well since it needs to examine
-    // return value which could be an obj ref.
+// At this point return value is computed and stored in RAX or XMM0.
+// On Amd64, Leave callback preserves the return register.  We keep
+// RAX alive by not reporting as trashed by helper call.  Also note
+// that GC cannot kick-in while executing inside profiler callback,
+// which is a requirement of profiler as well since it needs to examine
+// return value which could be an obj ref.
+
+#if defined(UNIX_AMD64_ABI)
+    getEmitter()->emitIns_R_R(INS_mov, emitTypeSize(TYP_ULONG), REG_R12, REG_RDI);
+    getEmitter()->emitIns_R_R(INS_mov, emitTypeSize(TYP_ULONG), REG_R13, REG_RSI);
+#endif
```

Comment:
Why do we need to save rdi and rsi?

At this point method no longer requires/cares about callee trash registers (other than return regs), why can't we use any callee trash registers other than return regs for profiler leave callback?